synapse/lib/ast.py

```diff
@@ -3239,13 +3239,20 @@ async def getPropAndValu(self, runt, path):
         if not path:
             return None, None
 
-        name = await self.kids[0].compute(runt, path)
+        propname = await self.kids[0].compute(runt, path)
+        name = await tostr(propname)
 
         ispiv = name.find('::') != -1
         if not ispiv:
 
             prop = path.node.form.props.get(name)
             if prop is None:
+                if not isinstance(propname, str):
```

Comment:
Should we functionalize this block to minimize copy/pasta?

Reply:
Added a `typeerr` function to stormtypes, open to other naming suggestions

Reply:
I think `typeerr` is fine, but I could also see it being something like `reqType` since it will raise.

Reply:
Yeah I initially had it as `reqtype`, but it doesn't raise the exception itself, it returns it so we can add the relevant AST info here.